code/controllers/subsystem/timer.dm

```diff
@@ -291,18 +302,19 @@ SUBSYSTEM_DEF(timer)
 		// Insert the timer into the bucket, and perform necessary circular doubly-linked list operations
 		new_bucket_count++
 		var/bucket_pos = BUCKET_POS(timer)
+		timer.bucket_pos = bucket_pos
+
 		var/datum/timedevent/bucket_head = bucket_list[bucket_pos]
 		if (!bucket_head)
 			bucket_list[bucket_pos] = timer
 			timer.next = null
 			timer.prev = null
 			continue
-		if (!bucket_head.prev)
-			bucket_head.prev = bucket_head
+
+		bucket_head.prev = timer
 		timer.next = bucket_head
-		timer.prev = bucket_head.prev
-		timer.next.prev = timer
-		timer.prev.next = timer
+		timer.prev = null
+		bucket_list[bucket_pos] = timer
```

Comment:
>Rewrote the circular linked list to a regular list, because it could cause timers to loop infinitely.

you should choose your words more carefully next time. this is not a "regular list" but a "regular linked list" (since arrays in byond are called `lists` i thought you moved this to use list() for the inner items of the bucket) 

(You technically converted this from a circular doubly linked list to a doubly linked list, since linked lists normally only have a head and a next)

Reply:
Thank you for pointing out the misleading description! Corrected.